File: Longest Common Prefix/longest_common_prefix.py
from copyreg import constructor
from sys import prefix


class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:     
        commonPrefix = min(strs, key=len)
        for word in strs: 
            prefix = str('')
            for firstWordChar, secondWordChar in zip(word, commonPrefix):
                if firstWordChar == secondWordChar: 
                    prefix += firstWordChar
                else:
                    commonPrefix = prefix
                    break        
            commonPrefix = prefix
        return commonPrefix

classInstance = Solution()
# print(classInstance.longestCommonPrefix(["dog","racecar","car"]))
# print(classInstance.longestCommonPrefix(["flower","flow","flight"]))
print(classInstance.longestCommonPrefix(["ab", "a"]))
# print(classInstance.longestCommonPrefix(["cir","car"]))

File: Longest Common Prefix/longest_common_prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
 var longestCommonPrefix = function(strs) {
    let commonPrefix = ''
    for(let i = 0; i<strs[0].length; i++) {
        for(let word of strs) {
            if(word[i] !== strs[0][i])
                return commonPrefix
        }   
        commonPrefix += strs[0][i]            
    }
    return commonPrefix
};

console.log(longestCommonPrefix(["dog","racecar","car"]))
console.log(longestCommonPrefix(["flower","flow","flight"]))